type_description_interfaces/msg/FieldType.msg

```diff
@@ -0,0 +1,149 @@
+# Represents the type of a field and related meta-data.
+
+# A constant for each type supported according to:
+#   http://design.ros2.org/articles/legacy_interface_definition.html
+# and:
+#   http://design.ros2.org/articles/idl_interface_definition.html
+# Order is loosely coupled to the order of appearance in the IDL 4.2 spec:
+#  https://www.omg.org/spec/IDL/4.2
+
+# Layout of constants across the 0-255 decimal values in the uint8:
+#
+# - 000-031: Primitive types, strings, and reserved space for future primitive types
+# - 032-063: Fixed sized array of primitive and string types
+# - 064-095: Bounded Sequences of primitive and string types
+# - 096-127: Unbounded Sequences of primitive and string types
+# - 128-255: Reserved space for future array/sequence-like types
+
+# Nested type defined in other .msg/.idl files.
+uint8 FIELD_TYPE_NESTED_TYPE = 0
+
+# Basic Types
+# Integer Types
+uint8 FIELD_TYPE_INT8 = 1
+uint8 FIELD_TYPE_UINT8 = 2
+uint8 FIELD_TYPE_INT16 = 3
+uint8 FIELD_TYPE_UINT16 = 4
+uint8 FIELD_TYPE_INT32 = 5
+uint8 FIELD_TYPE_UINT32 = 6
+uint8 FIELD_TYPE_INT64 = 7
+uint8 FIELD_TYPE_UINT64 = 8
+
+# Floating-Point Types
+uint8 FIELD_TYPE_FLOAT = 9
+uint8 FIELD_TYPE_DOUBLE = 10
+uint8 FIELD_TYPE_LONG_DOUBLE = 11
+
+# Char and WChar Types
+uint8 FIELD_TYPE_CHAR = 12
+uint8 FIELD_TYPE_WCHAR = 13
+
+# Boolean Type
+uint8 FIELD_TYPE_BOOLEAN = 14
+
+# Byte/Octet Type
+uint8 FIELD_TYPE_BYTE = 15
+
+# String Types
+uint8 FIELD_TYPE_STRING = 16
+uint8 FIELD_TYPE_WSTRING = 17
+
+# Fixed String Types
+uint8 FIELD_TYPE_FIXED_STRING = 18
```

Comment:
Note: there is no rosidl_parser.definition for "Fixed String". You can do fixed-size array of char, but this precludes "Sequences of Fixed String"s

Reply:
NOTE: this isn't an issue for translating from IDL->TypeDescription, but it could be a problem for the reverse

Reply:
I think we just leave this and if we encounter this issue (going from TypeDescription.msg to .idl and this option is used), we just throw saying it is not supported.